### DISCUSSION TITLE
Is there a way i can access the pathname in the `page.tsx` file in next13 with the new app folder?

### DISCUSSION THREAD

**Discussion by danielgroen** [vercel/next.js#46984] [Help]
### Summary

why do i want this?

for example, i have a sidebar. The sidebar i only need to render as ssr since the sidebar won't change.
Only the links inside the sidebar need a active class when the current path is the same as the `href` in the link component in the sidebar.

### Additional information

```js
So you have the sidebar here.
It would be helpfull if i have access to pathname, or just simply to the req object.

page.tsx


export default function layout() {
 return (
 <div className="sidebar">
  <Link href="/about">about</Link>
 </div>
 )
}
```

**Comment by vixero-dev** [comment]
Does [this](https://beta.nextjs.org/docs/api-reference/use-pathname) help?

**Comment by haydenbleasel** [comment]
Hi @danielgroen, you need to use a client component wrapper. I wrote an example on how to do this (with Tailwind too) here: https://www.beskar.co/blog/tailwind-active-next-13
